ably/options_test.go

```diff
@@ -1,57 +1,44 @@
 package ably_test
 
 import (
-	"log"
 	"net/url"
 
 	"github.com/ably/ably-go/ably"
 
 	. "github.com/ably/ably-go/Godeps/_workspace/src/github.com/onsi/ginkgo"
 	. "github.com/ably/ably-go/Godeps/_workspace/src/github.com/onsi/gomega"
-	"github.com/ably/ably-go/Godeps/_workspace/src/github.com/onsi/gomega/gbytes"
 )
 
 var _ = Describe("ClientOptions", func() {
-	var (
-		options *ably.ClientOptions
-		buffer  *gbytes.Buffer
-	)
+	var opts *ably.ClientOptions
 
 	BeforeEach(func() {
-		buffer = gbytes.NewBuffer()
+		opts = &ably.ClientOptions{}
+	})
 
-		options = &ably.ClientOptions{
-			Key: "id:secret",
-		}
+	Context("when Key is valid", func() {
+		var err error
 
-		options.Prepare()
-	})
+		BeforeEach(func() {
+			err = opts.SetKey("name:secret")
```

Comment:
This test-case tests options, not RestClient, I'm not convinced we should mixed responsibilities in the tests. However I'm going to revisit this tests after final modifications to this PR are applied according to other discussions.